airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -208,6 +209,9 @@ def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
           If the stream has no primary keys, return None.
         """
 
+    def generate_partitions(self, sync_mode: SyncMode, cursor_field: Optional[List[str]]) -> Iterable[Optional[Mapping[str, Any]]]:
```

Comment:
Why do we need this method? It does not seem to had anything in terms of concurrency